src/Symfony/Bundle/SecurityBundle/Resources/config/security.xml

```diff
@@ -21,11 +21,14 @@
         </service>
         <service id="Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface" alias="security.authorization_checker" />
 
-        <service id="security.token_storage" class="Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage" public="true">
+        <service id="security.token_storage" class="Symfony\Component\Security\Core\Authentication\Token\Storage\LazyTokenStorage" public="true">
```

Comment:
how about implementing a decoration for the `actual_token_storage` here, instead ? So that you can rename it `token_storage` and let this service decorate the token_storage ?